languages/ruby/exercises/concept/basics/.docs/introduction.md

```diff
@@ -0,0 +1,55 @@
+Ruby is a dynamic [object-oriented language]. Everything in Ruby is an [object][object].
+
+There are two ways to assign objects to names in Ruby - assigning variables or constants. Variables always start with lower-case letters and use [snake case][snake-case] for their formatting. A variable can be redefined to different objects over its lifetime. For example, `my_first_variable` can be defined and redefined many times using the `=` operator:
+
+```ruby
+my_first_variable = 1
+my_first_variable = "Some string"
+my_first_variable = SomeComplexObject.new
+```
+
+Constants, however, can only be assigned once. They must start with capital letters and are normally written in block capitals with words separated by underscores. For example:
+
+```ruby
+MY_FIRST_CONSTANT = 10
+
+# Redefining not allowed
+# MY_FIRST_CONSTANT = "Some String"
+```
+
+Ruby is organised into classes. Classes are defined using the `class` keyword followed by the name of the class. Classes are initialized using the `.new` method.
```

Comment:
```suggestion
Ruby is organised into classes. Classes are defined using the `class` keyword followed by the name of the class. Classes are initialized, or _instantiated_, using the `new` method.
```

Reply:
Why do we need underscores for the `instantiated`?

Reply:
I think we should choose either initialized or instantiated as the term we want to consistently use, rather than using both. I don't mind either term - whichever Ruby docs use would be best if anyone has a reference?

Reply:
I agree with one or another.  If it is introductory, it may be OK to define via context one using the other.  Since we use initialize to instantiate an object, I believe the term is used to describe how we get an instance of a class.

Reply:
This brings up the focus as well, Classes are initialized using `class`, while we get instances of that class by using `new`.  Perhaps the subject in that last sentence is the wrong noun.

Reply:
We could use:
"Instances of a class are instantiated using the `new` method"
But I think I'd prefer:
"Classes are instantiated using the `new` method."

Reply:
@pvcarrera Yeah - do that second one.

Reply:
Objects are instantiated by using the `new` method, classes are instantiated in other ways.

We create a class usually by doing

```ruby
class MyClass
end
```
This "instantiates a class" by creating a new object called `MyClass`.  We should be careful of mixing terms.  We instantiate objects of the class `MyClass` by doing `MyClass.new`

The instantiation of a class can be done by doing `MyClass = Class.new` which would be applying the idea directly, using the same terms as we would use when we generally talk about objects in Ruby.

Reply:
```suggestion
Ruby is organised into classes. Classes are defined using the `class` keyword followed by the name of the class. Objects are created by instantiating classes using the `.new` method. For example:
```

Reply:
"Objects are created by instantiating classes" is true... but it is only sometimes that the `.new` method is used.  This is immediately problematic as we start instantiating strings in examples, such as `fruit = 'banana'` where we have just instantiated a String object without using `.new`.  Or even the lie, when we use value objects, such as `the_answer = 42` and we get an `undefined method `new` for Integer:class`.

Instead:

    Objects are sometimes created by instnatiating classes using the `.new` method.